### PR TITLE
feat(ec2): SSH Key Import to ec2 instances

### PR DESCRIPTION
**Pull request by kshychko**
## Motivation and Context

per hamlet-io/engine#1489

## How Has This Been Tested?

On ssh segment for integration environment of one of the tenants, confirmed to be working as expected.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions

- [x] None

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] None of the above.

### CODE REVIEW

File: aws/components/bastion/setup.ftl [hunk 1]
```diff
@@ -148,6 +148,22 @@
                             (linkTargetAttributes.ACCESS_POINT_ID)!""
                         )]
                     [#break]
+                [#case USER_COMPONENT_TYPE]
```
[the remark this answers is not in the document]
Comment (roleyfoley) [reply aws/components/bastion/setup.ftl:151]: Yeah That would be a great candidate for a function. best spot is probably `services/ec2/resource.ftl`

File: aws/components/user/state.ftl [hunk 2]
```diff
@@ -54,7 +54,8 @@
                 "user" : {
                     "Id" : userId,
                     "Name" : core.ShortFullName,
-                    "Type" : AWS_IAM_USER_RESOURCE_TYPE
+                    "Type" : AWS_IAM_USER_RESOURCE_TYPE,
+                    "Deployed" : true
```
[the remark this answers is not in the document]
Comment (roleyfoley) [reply aws/components/user/state.ftl:58]: I'm not sure why that is there, feel free to remove it

[the remark this answers is not in the document]
Comment (kshychko) [reply aws/components/user/state.ftl:58]: @roleyfoley, without it the link isn't returned by `getLinkTargets` without `activeOnly` set to false, see https://github.com/hamlet-io/engine-plugin-aws/pull/218/files#diff-3796184d008a2a193301bd9b5c0845fbd8ce508d70dea7659ac8c89a5a239cbbR92
I don't think we should get all links here, including non-active ones, so my assumption is that we need to set this `Deployed` flag, as the user stack is never getting actually deployed. Please correct me, if I'm wrong.

File: aws/components/bastion/setup.ftl (hunk 1 above)
[the remark this answers is not in the document]
Comment (kshychko) [reply aws/components/bastion/setup.ftl:152]: @roleyfoley , done, please check out https://github.com/hamlet-io/engine-plugin-aws/pull/218/files#diff-f161ab09793941035a3061cc20a021fbf0ff460163821b593cd5505e1023a7eaR675

File: aws/services/ec2/resource.ftl [hunk 3]
```diff
@@ -672,6 +672,49 @@
...
+            [#if (SSHPublicKeys?keys)?seq_index_of(id) != ((SSHPublicKeys?keys)?size - 1)]
+                [#local SSHPublicKeysContent += "\n"]
+            [/#if]
+        [/#if]
+    [/#list]
+    [#return
+        {
+            "${priority}_scripts" : {
+                "files" :{
+                    "/home/ec2-user/.ssh/authorized_keys2" : {
```
Comment (kshychko) [inline aws/services/ec2/resource.ftl:689]: @roleyfoley , is it OK location for our additional file? Or should we store it somewhere under `/opt/codeontap` ? And what about the filename?  `authorized_keys2` has been deprecated, probably we should use something more specific?

File: aws/services/ec2/resource.ftl [hunk 4]
```diff
@@ -672,6 +672,49 @@
...
+                        "owner" : "ec2-user"
+                    }
+                },
+                "commands": {
+                    "01UpdateSSHDConfig" : {
+                        "command" : "sed -i 's#^\\(AuthorizedKeysFile.*$\\)#\\1 .ssh/authorized_keys2#' /etc/ssh/sshd_config",
+                        "ignoreErrors" : ignoreErrors
+                    },
+                    "02RestartSSHDService" : {
+                        "command" : "sudo service sshd restart",
```
Comment (roleyfoley) [inline aws/services/ec2/resource.ftl:709]: Is the sudo required? I thought CFN init ran as root.

**Review by roleyfoley — CHANGES_REQUESTED** [review]

File: aws/services/ec2/resource.ftl (hunk 4 above)
Comment (kshychko) [reply aws/services/ec2/resource.ftl]: fixed!

File: aws/services/ec2/resource.ftl (hunk 3 above)
Comment (roleyfoley) [reply aws/services/ec2/resource.ftl:689]: authorised keys are user based so I think its good to go into the ec2-user home drive. 
Maybe a better name for the file could be `authorized_keys_hamlet` just to show where the keys came from?

**Review by roleyfoley — CHANGES_REQUESTED** [review]

**Review by roleyfoley — APPROVED** [review]

**Pull request merged**